frame/3/gemm/bli_gemm_threading.c

```diff
@@ -112,13 +112,27 @@ void bli_gemm_thrinfo_free_paths( gemm_thrinfo_t** threads, dim_t num )
 gemm_thrinfo_t** bli_create_gemm_thrinfo_paths( )
 {
 
-#ifdef BLIS_ENABLE_MULTITHREADING 
+#ifdef BLIS_ENABLE_MULTITHREADING
     dim_t jc_way = bli_read_nway_from_env( "BLIS_JC_NT" );
-//    dim_t kc_way = bli_read_nway_from_env( "BLIS_KC_NT" );
+    // dim_t kc_way = bli_read_nway_from_env( "BLIS_KC_NT" );
     dim_t kc_way = 1;
     dim_t ic_way = bli_read_nway_from_env( "BLIS_IC_NT" );
     dim_t jr_way = bli_read_nway_from_env( "BLIS_JR_NT" );
     dim_t ir_way = bli_read_nway_from_env( "BLIS_IR_NT" );
+    char* str = getenv( "OMP_NUM_THREADS" );
+    if( str != NULL )
+    {
+       int max_threads = strtol( str, NULL, 10 );
+       max_threads = max( max_threads, 1 );
```

Comment:
You could also call it "crack" since I, for one, am addicted.

On 4/7/16 1:08 PM, Jeff Hammond wrote:
>
> In frame/3/gemm/bli_gemm_threading.c 
> <https://github.com/flame/blis/pull/62#discussion_r58917935>:
>
> >      dim_t kc_way = 1;
> >      dim_t ic_way = bli_read_nway_from_env( "BLIS_IC_NT" );
> >      dim_t jr_way = bli_read_nway_from_env( "BLIS_JR_NT" );
> >      dim_t ir_way = bli_read_nway_from_env( "BLIS_IR_NT" );
> > +    char* str = getenv( "OMP_NUM_THREADS" );
> > +    if( str != NULL )
> > +    {
> > +       int max_threads = strtol( str, NULL, 10 );
> > +       max_threads = max( max_threads, 1 );
>
> @poulson <https://github.com/poulson> Don't thank me, thank Travis: 
> https://travis-ci.org/flame/blis/builds/121480082 😉
>
> I'm going to start calling C++11 "chocolate C" in your honor 😆
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub 
> <https://github.com/flame/blis/pull/62/files/a0517dbf63048bbb1f677cffe84cd2396fe0b434#r58917935>
>